src/style/style.js

```diff
@@ -352,16 +351,22 @@ class Style extends Evented {
         } else {
             delete this.stylesheet.projection;
         }
-        this.updateProjection();
+        if (!this.map._explicitProjection) {
+            this.map._updateProjection(projection);
+        }
     }
 
-    updateProjection() {
-        const prevProjection = this.map.transform.projection;
-        const projectionChanged = this.map.transform.setProjection(this.map._runtimeProjection || (this.stylesheet ? this.stylesheet.projection : undefined));
-        const projection = this.map.transform.projection;
+    _updateProjection() {
+        if (!this.map._explicitProjection) {
+            this.map._updateProjection();
+        } else { // Ensure that style is consistent with current projection on style load
+            this.enableDraping();
+        }
```

Comment:
Should this logic/flow be part of and consolidated in `map._updateProjection`?

Reply:
I don't think so. This logic is called when there are changes in the style sheet. If the map has an explicit projection, it overrides the style spec projection, so no changes to the map are necessary.

This check is only necessary on style sheet updates.